docs/setup.md

```diff
@@ -88,108 +96,153 @@ IPU ES2K target.
+   ```bash
+   ./scripts/setup_infra.sh -i <8|16|..> -m <host> [-r <remote IP>]
+   ```
 
+   For SRIOV interfaces:
    ```bash
-   ./setup_infra.sh -i <8|16|..> -m <split|host> -r <10.10.0.2>
+   ./scripts/setup_infra_sriov.sh -i <8|16|..> -m <host>
    ```
 
    Where, the options:
      -i  Num interfaces to configure for deployment
-     -m  Mode host or split, depending on where Inframanager is configured to run
+     -m  Mode host for running inframanager on host
      -r  IP address configured by the user on the ACC-ARM complex for
        connectivity to the Host. This is provisioned using Node Policy - comms
-       channel ([5,0],[4,0]),([4,2],[0,3]). This is needed for runnning in split mode.
-       Script will assign an IP addresss from the same subnet on the Host side for connectivity.
-
-
-   Please set following env variables for host deployment:
-     SDE_INSTALL - Default SDE install directory
-     P4CP_INSTALL - Default p4cp install directory
+       channel ([5,0],[4,0]),([4,2],[0,3]). This is needed only for runnning
+       in split mode.
+
+   Script will auto assign an IP addresss from the same subnet on the Host side
+   vport for connectivity. The communication channel vport interface name is
+   autodetected by the script for the above mentioned comms channel configuration
+   in the cdq node policy.
+
+   Please also set following env variables for the deployment. These paths are
+   needed to set the dependencies correctly.
+     SDE_INSTALL - Default p4sde install directory
+     P4CP_INSTALL - Default p4-cp install directory
      DEPEND_INSTALL - Default target dependencies directory
-
+     K8S_RECIPE - Path to k8s recipe on the host
 
    After running the above script, verify that infrap4d is running.
+
    ```bash
    ps -ef | grep infrap4d
    ```
+
    ```none
    root     1254701       1 99 13:34 ?        00:13:10 /host/networking-recipe/install/sbin/infrap4d
    ```
 
-   On ES2K target, this script will also load the IDPF driver. Verify the
+   On ES2100 target, this script will also load the IDPF driver. Verify the
    presence of the PF:
+
    ```bash
    devlink dev show
    ```
+
    ```none
    pci/0000:af:00.0
    ```
 
-7. Run ARP-Proxy script, which creates a new namespace and assigns an interface
+7. For the Intel IPU ES2100, connect to IMC from host and run the following command on IMC:
+   ```bash
+   devmem 0x202920C100 64 0x8yy
+   ```
+   where yy is the 2nd octet of the MAC of the interface on host ending with 'd5' (when using CDQ) or 'v1' (when using SR-IOV).
+   For example if the interface `ens801f0d5` has MAC `00:11:00:05:03:14`, then it needs to be
+   ```bash
+   devmem 0x202920C100 64 0x811
+   ```
+
+   This command is currently required for the core-dns component of kubernetes to function.
+   For CDQ till d3 are default sub-functions so first interface d4 needs to be assigned to arp and d5 to host.
+   For SRIOV - Frist VF vf0 goes to arp and VF1 goes to host
```

Comment:
For sriov, it's v0 and v1. Can we also add "ending with" in front of these suffixes for more clarity. e.g. "interface ending with d3", "interface ending with v0", etc.